### PR TITLE
updated axis code to have an integer flag

### PR DESCRIPTION
**Pull request by abhishekpratapa**
references PR https://github.com/vega/vega-parser/pull/76

### CODE REVIEW

**Comment by domoritz** [comment]
Thank you for the PR. I wonder whether a more general solution would be to not show duplicate axis labels if they are adjacent. This way, one could use multiples beyond just integers or other kinds of formatting as well. I don't know exactly what this would look like but it would be a more general solution and we wouldn't even need users to provide any additional arguments.

File: src/ticks.js [hunk 1]
```diff
@@ -75,8 +75,8 @@ export function validTicks(scale, ticks, count) {
  * @param {*} [count] - The approximate number of desired ticks.
  * @return {Array<*>} - The generated tick values.
  */
-export function tickValues(scale, count) {
-  return scale.ticks ? scale.ticks(count) : scale.domain();
+export function tickValues(scale, count, integer) {
+  return scale.ticks ? scale.ticks((integer == null || !integer)?count:((scale.domain()[1] < count)?scale.domain()[1]:count)) : scale.domain();
```
Comment (jheer) [inline src/ticks.js:79]: This code is error prone. For example scale domains can have more than two elements. There is no guarantee that `domain[1]` is the maximum value. FWIW, I tend to use `peek(domain)` to pull last the array value.

File: src/AxisTicks.js [hunk 2]
```diff
@@ -32,10 +32,11 @@ prototype.transform = function(_, pulse) {
   var out = pulse.fork(pulse.NO_SOURCE | pulse.NO_FIELDS),
       ticks = this.value,
       scale = _.scale,
+      integer = _.integer,
```
Comment (jheer) [inline src/AxisTicks.js:35]: New parameters should be documented in the JS doc comment above.

**Comment by domoritz** [comment]
I described my alternative proposal in https://github.com/vega/vega-encode/issues/14.

**Comment by jheer** [comment]
Closing out this PR in favor of exploring other approaches.

**Pull request closed without merging**